huaweicloud/services/css/resource_huaweicloud_css_logstash_configuration.go

```diff
@@ -0,0 +1,315 @@
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	updateLogstashCnfChanges := []string{
+		"conf_content",
+		"setting",
+		"sensitive_words",
+	}
+
+	if d.HasChanges(updateLogstashCnfChanges...) {
+		updateCnfOpts := buildCreateLogstashCnfParameters(d)
+
+		_, err = cssV1Client.UpdateCnf(&model.UpdateCnfRequest{
+			ClusterId: d.Get("cluster_id").(string),
+			Body:      updateCnfOpts,
+		})
+		if err != nil {
+			diag.Errorf("error updating CSS logstash cluster configuration: %s", err)
+		}
+	}
+
+	checkErr := configFileStatusCheck(ctx, d, cssV1Client)
+	if checkErr != nil {
+		return diag.FromErr(checkErr)
+	}
+
+	return resourceLogstashConfigurationRead(ctx, d, meta)
+}
+
+func resourceLogstashConfigurationDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	cssV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	_, err = cssV1Client.DeleteConf(&model.DeleteConfRequest{
+		ClusterId: d.Get("cluster_id").(string),
+		Body: &model.DeleteConfReq{
+			Name: d.Get("name").(string),
+		},
+	})
+	if err != nil {
+		return diag.Errorf("error deleting CSS logstash cluster configuration: %s", err)
+	}
+
+	return nil
+}
+
+func configFileStatusCheck(ctx context.Context, d *schema.ResourceData, cssV1Client *cssv1.CssClient) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"checking"},
+		Target:  []string{"available", "unavailable"},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := cssV1Client.ShowGetConfDetail(&model.ShowGetConfDetailRequest{
+				ClusterId: d.Get("cluster_id").(string),
+				Name:      d.Get("name").(string),
+			})
+			if err != nil {
+				return nil, "failed", err
+			}
+			return resp, *resp.Status, err
+		},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        30 * time.Second,
+		PollInterval: 30 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for the CSS logstash cluster config to check completed: %s", err)
+	}
+	return nil
+}
+
+func resourceLogstashConfigurationImportState(_ context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	cssV1Client, err := conf.HcCssV1Client(region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CSS V1 client: %s", err)
+	}
+
+	parts := strings.Split(d.Id(), "/")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format of import ID, must be <cluster_id>/<name>")
+	}
+
+	d.Set("cluster_id", parts[0])
+	d.Set("name", parts[1])
+
+	_, err = cssV1Client.ShowGetConfDetail(&model.ShowGetConfDetailRequest{
+		ClusterId: d.Get("cluster_id").(string),
+		Name:      d.Get("name").(string),
+	})
+	if err != nil {
+		return []*schema.ResourceData{d}, err
+	}
```

Comment:
it seems useless, the read method will be called when importing the resouce, this func will be called